File: frontend/src/objects/admin/afterSales/AdminTicketIdentity.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  OrderId,
  SummaryText,
  TicketId,
  TicketKind,
  TicketStatus,
} from '@/objects/core/SharedObjects'

export type AdminTicketIdentity = {
  id: TicketId
  orderId: OrderId
  kind: TicketKind
  status: TicketStatus
  summary: SummaryText
}

File: backend/src/main/scala/services/admin/objects/AdminTicketIdentity.scala
package services.admin.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import services.order.objects.*

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class AdminTicketIdentity(
    id: TicketId,
    orderId: OrderId,
    kind: TicketKind,
    status: TicketStatus,
    summary: SummaryText,
)
object AdminTicketIdentity:
  given Encoder[AdminTicketIdentity] = deriveEncoder
  given Decoder[AdminTicketIdentity] = deriveDecoder
